### PR TITLE
Launch image builder based on image version

### LINKED ISSUE #154: Image-builder launch only when there are a new kubernetes release
### Explain problem to solve

For the moment, kubernetes image is launch each sunday but we don't change kubernetes version.
(the current project only manage one version at a time.
packer script:
https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/scripts/ci-outscale-nightly.sh
overide-json:
https://github.com/kubernetes-sigs/image-builder/blob/4d74d81b1c0bba3ec5b2493297c5644a4a799560/images/capi/packer/outscale/ci/nightly/overwrite-1-24.json

### Describe the solution you would like

Options 1:
Detect new version of kubernetes release, create a new pr on image builder to change version.

Options 2:
Detect new version of kubernetes release, change json file with new version and create a pr from time to time to image-builder.

### Additional context

None

### Environment
```
- Kubernetes version: (use `kubectl version`): v1.22.5-rancher1-1
- OS (e.g. from `/etc/os-release`): CentOS Linux 7 (Core)
- Kernel (e.g. `uname -a`): 3.10.0-1160.15.2.el7.x86_64
- cluster-api-provider-outscale version: v1.1.3
- cluster-api version: current
- Install tools: calico, ingress-nginx, osc-ccm
- Kubernetes Distribution: rke
- Kubernetes Distribution version: v1.3.4
```

### PR DESCRIPTION
**Pull request by outscale-vbr**
**What type of PR is this?**
/kind feature

 

**What this PR does / why we need it**:
It add github action to launch the creation of newest image trigger by event.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #154

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

File: .github/workflows/dispatch_image_builder.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        ref: add-outscale-public-image
+    - uses: actions/setup-go@v3
+      with:
+        go-version: '1.18.7'
+    - uses: actions/checkout@v3
+      with:
+        repository: outscale-dev/cluster-api-provider-outscale
+        ref: main
+        path: cluster-api-provider-outscale
+        token: ${{ secrets[matrix.github_token] }}
```
Comment (outscale-mdr) [inline .github/workflows/dispatch_image_builder.yaml:54]: why a second checkout ?

File: .github/workflows/dispatch_image_builder.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      run: make install-packer
+      shell: bash
+      working-directory: ${{ github.workspace }}/cluster-api-provider-outscale
+    - name: generate file
+      shell: bash
+      run: |
+        kubernetes_series=`echo ${{ matrix.version }} | cut -d "." -f1-2`
+        kubernetes_semver=`echo ${{ matrix.version }}`
+        ./json_format -r ${kubernetes_semver} -v ${kubernetes_series} -o overwrite-k8s.json 
+        cat overwrite-k8s.json 000
```
Comment (outscale-mdr) [inline .github/workflows/dispatch_image_builder.yaml:65]: 000 ?

File: .github/workflows/dispatch_image_builder.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
+on:
+  repository_dispatch:
+    types:
+      - release
```
Comment (outscale-mdr) [inline .github/workflows/dispatch_image_builder.yaml:4]: Will you remove https://github.com/outscale/cluster-api-provider-outscale/blob/main/.github/workflows/image-builder.yaml ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

File: .github/workflows/dispatch_image_builder.yaml (hunk 3 above)
Comment (outscale-vbr) [reply .github/workflows/dispatch_image_builder.yaml:4]: Yes

File: .github/workflows/dispatch_image_builder.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            osc_secret_key: OSC_OPENSOURCE_SECRET_KEY
+            osc_region: OSC_OPENSOURCE_REGION
+            osc_account_id: OSC_OPENSOURCE_ACCOUNT_ID
+            github_token: SECRET_GITHUB_TOKEN    
+          - environment: cloudgouv-eu-west-1
+            osc_access_key: OSC_SEC1_ACCESS_KEY
+            osc_secret_key: OSC_SEC1_SECRET_KEY
+            osc_region: OSC_SEC1_REGION
+            osc_account_id: OSC_SEC_ACCOUNT_ID
+            github_token: SECRET_GITHUB_TOKEN
```
[the remark this answers is not in the document]
Comment (outscale-vbr) [reply .github/workflows/dispatch_image_builder.yaml]: Ok i replace with environment

File: .github/workflows/dispatch_image_builder.yaml (hunk 1 above)
Comment (outscale-vbr) [reply .github/workflows/dispatch_image_builder.yaml]: It is not useful.

File: .github/workflows/dispatch_image_builder.yaml (hunk 2 above)
Comment (outscale-vbr) [reply .github/workflows/dispatch_image_builder.yaml]: Thanks

File: .github/workflows/dispatch_image_builder.yaml (hunk 3 above)
Comment (outscale-vbr) [reply .github/workflows/dispatch_image_builder.yaml:4]: Yes

**Comment by outscale-vbr** [comment]
https://github.com/outscale/cluster-api-provider-outscale/pull/221 will be merge first to handle cve.
and change on trivyignore will be removed for this PR.

File: .github/workflows/dispatch_image_builder.yaml (hunk 3 above)
Comment (outscale-vbr) [reply .github/workflows/dispatch_image_builder.yaml:4]: I will remove it in an other PR (if you ok ?) when there will be a new release of kubernetes 1.26.2.

Comment (outscale-mdr) [reply .github/workflows/dispatch_image_builder.yaml:4]: Ok 👍

File: .trivyignore [hunk 5]
```diff
@@ -1,4 +0,0 @@
-CVE-2020-8912
-CVE-2020-8911 
-GHSA-7f33-f4f5-xwgw
-GHSA-f5pg-7wfw-84q9
```
Comment (outscale-mdr) [inline .trivyignore:4]: What is it for ? (build error #221) ?

Comment (outscale-vbr) [reply .trivyignore:4]: Yes

**Comment by outscale-vbr** [comment]
@outscale-mdr Are you ok with this PR ?

**Pull request merged**